### ISSUE TITLE
Implement a Class Factory

### ISSUE THREAD

**Issue by ZaxR** [ZaxR/bulwark#1]
The purpose of this factory would be to remove the need to manually create a Class for each function to create the decorator. Start of some potential code:

```
from inspect import getmembers, isfunction
import sys

import bulwark.checks as ck


def snake_to_camel(snake_str):
    components = snake_str.split('_')
    return ''.join(x.title() for x in components)


def decorator_factory(decorator_name, func):
    """Takes in a function and outputs a class that can be used as a decorator."""
    class decorator_name(BaseDecorator):
        check_func = staticmethod(func)

    return decorator_name


# Automatically creates decorators for each function in bulwark.checks
this_module = sys.modules[__name__]
check_functions = getmembers(ck, isfunction)

for func in check_functions:
    decorator_name = snake_to_camel(func[1].__name__)
    setattr(this_module, decorator_name, decorator_factory(decorator_name, func))


# issue with this is that it includes functions ck imported
```

**Comment by ZaxR** [comment]
check_functions = [func[1] for func in getmembers(ck, isfunction) if func[1].__module__ == 'bulwark.checks']

**Comment by ZaxR** [comment]
Closed by #2
